request/src/main/java/com/vimeo/networking2/Configuration.kt

```diff
@@ -0,0 +1,272 @@
+ * client.
+ * @param requestTimeoutSeconds The maximum number of seconds a request will wait to complete before timing out.
+ * @param isCertPinningEnabled True if the library should pin certificates, false otherwise. Note: should be turned off
+ * when debugging.
+ * @param logDelegate The instance to which logging will be delegated.
+ * @param logLevel The level of logging which will be performed by the library.
+ * @param cacheDirectory The optional directory in which requests will be cached.
+ * @param cacheMaxSizeBytes The maximum size of the cache, if one was provided.
+ * @param cacheMaxAgeSeconds The maximum age of items in the cache, if one was provided.
+ */
+data class Configuration(
+    val baseUrl: String,
+
+    val clientId: String,
+    val clientSecret: String,
+    val scope: Scopes,
+    val codeGrantRedirectUri: String,
+
+    val locales: List<Locale>,
+
+    val accountStore: AccountStore,
+
+    val networkInterceptors: List<Interceptor>,
+    val interceptors: List<Interceptor>,
+
+    val userAgent: String,
+
+    val requestTimeoutSeconds: Int,
+
+    val isCertPinningEnabled: Boolean,
+
+    val logDelegate: LogDelegate?,
+    val logLevel: LogDelegate.Level,
+
+    val cacheDirectory: File?,
+    val cacheMaxSizeBytes: Long,
+    val cacheMaxAgeSeconds: Int
+) {
+
+    /**
+     * The builder for the [Configuration].
+     *
+     * @param clientId The Vimeo API client ID, required to construct an instance.
+     * @param clientSecret The Vimeo API client secret, required to construct an instance.
+     * @param scope The permission scopes requested by the client, required to construct an instance.
+     */
+    class Builder(private val clientId: String, private val clientSecret: String, private val scope: Scopes) {
+        private var baseUrl: String = DEFAULT_BASE_URL
+
+        private var codeGrantRedirectUrl: String = "vimeo$clientId://auth"
+
+        private var locales: List<Locale> = listOf(Locale.US)
+
+        private var accountStore: AccountStore = InMemoryAccountStore()
+
+        private var networkInterceptors: List<Interceptor> = emptyList()
+        private var interceptors: List<Interceptor> = emptyList()
+
+        private var userAgent: String = DEFAULT_USER_AGENT
+
+        private var requestTimeoutSeconds: Int = DEFAULT_TIMEOUT
+
+        private var isCertPinningEnabled: Boolean = true
+
+        private var logDelegate: LogDelegate? = DefaultLogDelegate()
+        private var logLevel: LogDelegate.Level = LogDelegate.Level.DEBUG
+
+        private var cacheDirectory: File? = null
+        private var cacheMaxSizeBytes: Long = DEFAULT_CACHE_SIZE
+        private var cacheMaxAgeSeconds: Int = DEFAULT_CACHE_MAX_AGE
+
+        /**
+         * Specify a base URL. Defaults to [DEFAULT_BASE_URL].
+         *
+         * @see Configuration.baseUrl
+         */
+        fun withBaseUrl(baseUrl: String) = apply { this.baseUrl = baseUrl }
```

Comment:
I _think_ we already talked about this but I like the choice to go with a `with` prefix over a `set` prefix for the `Builder` methods.  👏